### PR TITLE
faster when mask is all true

### PR DESCRIPTION
**Pull request by ahnitz**

### CODE REVIEW

File: pycbc/io/hdf.py [hunk 1]
```diff
@@ -641,12 +641,11 @@ def newsnr_sgveto_psdvar(self):
                                            self.sgchisq, self.psd_var_val)
 
     def get_column(self, cname):
-        if self.mask is not None:
+        if self.mask is not None and (self.mask.sum() != len(self.mask)):
```
Comment (spxiwh) [inline pycbc/io/hdf.py]: It might be worth doing things the old way if `self.mask.sum()` is at least 30% of the full set (say)

Comment (ahnitz) [reply pycbc/io/hdf.py]: Yeah, that would make sense.

File: pycbc/io/hdf.py [hunk 2]
```diff
@@ -641,12 +641,14 @@ def newsnr_sgveto_psdvar(self):
                                            self.sgchisq, self.psd_var_val)
 
     def get_column(self, cname):
-        if self.mask is not None:
+        MFRAC = 0.3
```
Comment (tdent) [inline pycbc/io/hdf.py]: please put a comment explaining what is happening here (ie why the differences). 
also, the logic would be simpler if the `self.mask is None` case was done first. 
also/however it seems like the init for the class results in self.mask *never* being None, so I'm not sure why there is still a test for `self.mask is not None` here.

Comment (tdent) [reply pycbc/io/hdf.py]: also, is there any evidence for the value of MFRAC -- i.e. what's the largest fraction of triggers for which you have successfully used the direct selection `self.trigs[cname][self.mask]` ?

**Comment by tdent** [comment]
Have tested this making singles plots on some large (1-1.5e9 triggers) trigger merge files, it seems to be OK.  I will ask for some comments to be added and maybe rearrange the logic to be clearer.

File: pycbc/io/hdf.py (hunk 2 above)
Comment (tdent) [reply pycbc/io/hdf.py]: Please replace the h5py selection in `def template_id(self):` with a call to `get_column()` as for all the other datasets present in trigs.

**Comment by tdent** [comment]
@ahnitz ping on this.

**Comment by spxiwh** [comment]
Can/should I merge this?

**Comment by tdent** [comment]
Well ... no, since there was no response yet to the changes I requested. See the big 'review required' red cross. The `template_id` method needs to be fixed, for one.

**Comment by tdent** [comment]
@ahnitz this still requires action (still urgent from my point of view). If something is stopping you from addressing it I can try and make edits in the browser.

**Comment by ahnitz** [comment]
@tdent Sorry Tom for the delay, I should be able to handle this in the morning.

File: pycbc/io/hdf.py (hunk 2 above)
Comment (ahnitz) [reply pycbc/io/hdf.py]: @tdent I'm not sure the largest MFRAC that will work, most test cases involve it being lest than 10% of the entire array. The current value is a guess.

**Pull request merged**